File: src/brs/http/GetAccountsWithName.java
package brs.http;

import brs.Account;
import brs.BurstException;
import brs.db.BurstIterator;
import brs.services.AccountService;
import brs.util.Convert;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;

import static brs.http.common.Parameters.ACCOUNTS_RESPONSE;
import static brs.http.common.Parameters.NAME_PARAMETER;

;
;

public class GetAccountsWithName extends APIServlet.APIRequestHandler {

    private final AccountService accountService;

    GetAccountsWithName(AccountService accountService) {
        super(new APITag[] {APITag.ACCOUNTS}, NAME_PARAMETER);
        this.accountService = accountService;
    }

    @Override
    JsonElement processRequest(HttpServletRequest request) throws BurstException {
        BurstIterator<Account> accounts = accountService.getAccountsWithName(request.getParameter(NAME_PARAMETER));
        JsonArray accountIds = new JsonArray();

        while (accounts.hasNext()) {
            accountIds.add(Convert.toUnsignedLong(accounts.next().id));
        }

        accounts.close();

        JsonObject response = new JsonObject();
        response.add(ACCOUNTS_RESPONSE, accountIds);
        return response;
    }
}

File: src/brs/http/GetAccountsWithName.kt
package brs.http

import brs.Account
import brs.BurstException
import brs.services.AccountService
import brs.util.Convert
import com.google.gson.JsonArray
import com.google.gson.JsonElement
import com.google.gson.JsonObject

import javax.servlet.http.HttpServletRequest

import brs.http.common.Parameters.ACCOUNTS_RESPONSE
import brs.http.common.Parameters.NAME_PARAMETER

internal class GetAccountsWithName internal constructor(private val accountService: AccountService) : APIServlet.JsonRequestHandler(arrayOf(APITag.ACCOUNTS), NAME_PARAMETER) {

    @Throws(BurstException::class)
    internal override fun processRequest(request: HttpServletRequest): JsonElement {
        val accounts = accountService.getAccountsWithName(request.getParameter(NAME_PARAMETER))
        val accountIds = JsonArray()

        for (account in accounts) {
            accountIds.add(Convert.toUnsignedLong(account.id))
        }

        val response = JsonObject()
        response.add(ACCOUNTS_RESPONSE, accountIds)
        return response
    }
}
